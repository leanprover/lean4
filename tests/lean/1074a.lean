inductive Term
| id2: Term -> Term -> Term

inductive Brx: Term -> Prop
| id2: Brx z -> Brx (Term.id2 n z)

def Brx.interp {a} (H: Brx a): Nat :=
  match a with
  | Term.id2 n z => by
    let ⟨Hn, Hz⟩: True ∧ Brx z
      := by cases H <;> exact ⟨by simp, by assumption⟩;
    exact Hz.interp

def Brx.interp_nil (H: Brx a): H.interp = H.interp
  :=
  by {
    unfold interp
    rfl
  }

#check Brx.interp._eq_1
