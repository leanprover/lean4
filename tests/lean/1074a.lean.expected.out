Brx.interp._eq_1 : ∀ (n z : Term) (H_2 : Brx (Term.id2 n z)),
  Brx.interp H_2 =
    match (_ : True ∧ Brx z) with
    | (_ : True ∧ Brx z) => Brx.interp Hz
